### PR TITLE
Add more unit tests

### PR DESCRIPTION
**Pull request by matt-frey**
This pull request fixes a bug in the calculation of the ellipsoid points and adds following unit tests:
* test_ellipsoid_reflection.f90
* test_grid2par.f90
* test_free_slip_3d.f90

### CODE REVIEW

File: src/3d/parcels/parcel_ellipsoid.f90 [hunk 1]
```diff
@@ -180,15 +180,32 @@ function get_ellipsoid_points(position, volume, B) result(points)
...
                 ! suppport point in the global reference frame
                 do k = 1, 3
-                    points(j, k) = position(k)            &
-                                 + points(j, 1) * V(k, 1) &
-                                 + points(j, 2) * V(k, 2) &
-                                 + points(j, 3) * V(k, 3)
+                    points(j, k) = position(k)        &
+                                 + point(1) * V(k, 1) &
+                                 + point(2) * V(k, 2) &
+                                 + point(3) * V(k, 3)
```
Comment (matt-frey) [inline src/3d/parcels/parcel_ellipsoid.f90:190]: In principle we can remove this line, since `points(3)` is always zero.

Comment (sjboeing) [reply src/3d/parcels/parcel_ellipsoid.f90:190]: The naming "points" and "point" may be confusing, are there more explicit names?

File: src/3d/parcels/parcel_interpl.f90 [hunk 2]
```diff
@@ -352,13 +348,13 @@ subroutine grid2par(vel, vor, vgrad, add)
...
 
                         ! loop over field components
-                        do c = 1, ncomp
+                        do c = 1, 3
                             ! the weight is a quarter due to 4 points per ellipsoid
                             vel(n, c) = vel(n, c) &
                                       + weight * velog(ks(l), js(l), is(l), c)
                         enddo
 
-                        do c = 1, 4
```
Comment (sjboeing) [inline src/3d/parcels/parcel_interpl.f90:361]: Is velgradg now 5 elements?

File: src/3d/parcels/parcel_interpl.f90 [hunk 3]
```diff
@@ -352,13 +348,13 @@ subroutine grid2par(vel, vor, vgrad, add)
...
+                        do c = 1, 9
```
Comment (sjboeing) [inline src/3d/parcels/parcel_interpl.f90:357]: Is velgradg now 5 elements?

File: unit-tests/3d/test_free_slip_3d.f90 [hunk 4]
```diff
@@ -0,0 +1,84 @@
+! =============================================================================
+!                       Test free slip boundary condition
+!
+!       This unit test checks the free slip boundary condition which is
+!       implied in vertical direction. Ellipses with aspect ratio 3.5
+!       are place such that the parcels at the bottom and ad the top have
```
Comment (sjboeing) [inline unit-tests/3d/test_free_slip_3d.f90:6]: "placed", "at"

File: unit-tests/3d/test_grid2par.f90 [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    call field_alloc
+
+    n_per_dim = 3
+
+    n_parcels = n_per_dim ** 3 * nx *ny *nz
+    call parcel_alloc(n_parcels)
+
+    allocate(vel(n_parcels, 3))
+    allocate(vor(n_parcels, 3))
+    allocate(vgrad(n_parcels, 9))
```
Comment (sjboeing) [inline unit-tests/3d/test_grid2par.f90:41]: Should this be 5 elements  now?

**Review by sjboeing — COMMENTED** [review]
Hi Matthias,

I had a brief look at this, but didn't check the details. Maybe we can talk about the last test briefly tomorrow.

File: unit-tests/3d/test_free_slip_3d.f90 (hunk 4 above)
Comment (matt-frey) [reply unit-tests/3d/test_free_slip_3d.f90:6]: Thanks.

File: src/3d/parcels/parcel_interpl.f90 (hunk 3 above)
Comment (matt-frey) [reply src/3d/parcels/parcel_interpl.f90:357]: Yes, it seems I missed this in #248.

File: src/3d/parcels/parcel_ellipsoid.f90 (hunk 1 above)
Comment (matt-frey) [reply src/3d/parcels/parcel_ellipsoid.f90:190]: We may use `xy` instead and remove the third dimension, since it is not necessary.

Comment (matt-frey) [reply src/3d/parcels/parcel_ellipsoid.f90]: I pushed my proposed change.

**Comment by matt-frey** [comment]
Thanks @sjboeing. I updated the PR.

**Comment by sjboeing** [comment]
@matt-frey: I am happy for you to merge: I haven't gone through them completely but since these are tests they are checking existing functionality. If you would like me to have a closer look, maybe we can look at it together.

**Comment by matt-frey** [comment]
@sjboeing Yes, they test existing functionality, but just ported to 3D. Sure, we can also have a look together,
